aten/src/ATen/native/DilatedConvolution.cpp

```diff
@@ -0,0 +1,1090 @@
+  return std::tie(grad_input, grad_weight, grad_bias);
+}
+
+std::tuple<Tensor, Tensor, Tensor> conv_dilated2d_backward_cpu(
+    const Tensor& grad_output,
+    const Tensor& input,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    IntArrayRef stride_size,
+    IntArrayRef pad_size,
+    IntArrayRef dilation_size,
+    const Tensor& columns,
+    const Tensor& ones,
+    const std::array<bool, 3ul> output_mask) {
+  auto options = grad_output.options();
+  Tensor grad_input = at::empty({0}, options);
+  Tensor grad_weight = at::empty({0}, options);
+  Tensor grad_bias = at::empty({0}, options);
+  CALL_BACKWARD_OUT(2);
+  return std::tie(grad_input, grad_weight, grad_bias);
+}
+
+std::tuple<Tensor&, Tensor&, Tensor&> conv_dilated3d_out_cpu(
+    Tensor& output,
+    Tensor& columns,
+    Tensor& ones,
+    const Tensor& input,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    const Tensor& bias,
+    IntArrayRef stride_size,
+    IntArrayRef pad_size,
+    IntArrayRef dilation_size) {
+  auto options = input.options();
+  Tensor grad_output = at::empty({0}, options); // not used
+  Tensor grad_input = at::empty({0}, options); // not used
+  Tensor grad_weight = at::empty({0}, options); // not used
+  Tensor grad_bias = at::empty({0}, options); // not used
+  int64_t nOutputPlane = weight.size(0);
+  int64_t outputDepth = OUTPUTSIZE(0, 3);
+  int64_t outputHeight = OUTPUTSIZE(1, 3);
+  int64_t outputWidth = OUTPUTSIZE(2, 3);
+  output.resize_({nOutputPlane, outputDepth, outputHeight, outputWidth});
+  if (input.dim() == 5) {
+    INSERT_BATCH_DIMENSION(output, input.size(0));
+  }
+  CALL_TEMPLATE(3);
+  return std::tie(output, columns, ones);
+}
+
+Tensor& conv_dilated3d_out_cpu(
+    Tensor& output,
+    const Tensor& input,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    const Tensor& bias,
+    IntArrayRef stride_size,
+    IntArrayRef pad_size,
+    IntArrayRef dilation_size) {
+  auto options = input.options();
+  Tensor columns = at::empty({0}, options);
+  Tensor ones = at::empty({0}, options);
+  CALL_OUT(3);
+  return output;
+}
+
+Tensor conv_dilated3d_cpu(
+    const Tensor& input,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    const Tensor& bias,
+    IntArrayRef stride_size,
+    IntArrayRef pad_size,
+    IntArrayRef dilation_size) {
+  auto options = input.options();
+  Tensor output = at::empty({0}, options);
+  Tensor columns = at::empty({0}, options);
+  Tensor ones = at::empty({0}, options);
+  CALL_OUT(3);
+  return output;
+}
+
+std::tuple<Tensor&, Tensor&, Tensor&> conv_dilated3d_forward_out_cpu(
+    Tensor& output,
+    Tensor& columns,
+    Tensor& ones,
+    const Tensor& input,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    const Tensor& bias,
+    IntArrayRef stride_size,
+    IntArrayRef pad_size,
+    IntArrayRef dilation_size) {
+  std::cout << "NOT IMPLEMENTED: conv_dilated3d_forward_out_cpu3" << std::endl;
+  CALL_OUT(3); // Is this correct??
```

Comment:
You tell me!

Reply:
yes, it's correct, the function itself may be dead though, see above.